src/overlays/actors/ovl_Bg_Bdan_Switch/z_bg_bdan_switch.c

```diff
@@ -90,17 +90,21 @@ static InitChainEntry sInitChain[] = {
 static Vec3f D_8086E0E0 = { 0.0f, 140.0f, 0.0f };
 
 void BgBdanSwitch_InitDynaPoly(BgBdanSwitch* this, PlayState* play, CollisionHeader* collision, s32 flag) {
-    s16 pad1;
+    s32 pad;
     CollisionHeader* colHeader = NULL;
-    s16 pad2;
 
     DynaPolyActor_Init(&this->dyna, flag);
     CollisionHeader_GetVirtual(collision, &colHeader);
     this->dyna.bgId = DynaPoly_SetBgActor(play, &play->colCtx.dyna, &this->dyna.actor, colHeader);
+
+#if OOT_DEBUG
     if (this->dyna.bgId == BG_ACTOR_MAX) {
+        s32 pad2;
+
         PRINTF("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_bdan_switch.c", 325,
                this->dyna.actor.id, this->dyna.actor.params);
     }
+#endif
```

Comment:
I think we want to handle these InitDynapoly functions globally / all at once (they're copypasted all over)
@fig02 ?

Reply:
Might need polish/better macro name but something like this?
https://github.com/zeldaret/oot/compare/main...engineer124:dyna_debug_print?expand=1

Reply:
Should the string "Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n" be part of the macro too? And I would probably inline the weird BgBombwall_InitDynapoly case instead of making a separate macro for it

Reply:
Ideally yes, but when I tried that before, I got rodata reordering and it wouldn't match

Reply:
Hmm I tried it just now and didn't get any reordering. But, it looks like the japanese text is being encoded as UTF-8 instead of EUC-JP

Reply:
Yeah, I think it works if you write the string like this:
```
"Warning : move BG \xc5\xd0\xcf\xbf\xbc\xba\xc7\xd4(%s %d)(name %d)(arg_data 0x%04x)\n"
```

Reply:
fwiw yeah if the string is in a header it wouldn't be reencoded from utf-8 to euc-jp
that's because asm-processor does the encoding conversion and it only processes the file it's given (+ any single include that follows a `#pragma asmproc recurse`)

Reply:
Im not really sure what to do about the string.
I do think that we can possibly fit even more of the init into the macro, like this stuff. and call it `DYNAPOLY_INIT_AND_SET()` or something along those lines
```c
    DynaPolyActor_Init(&this->dyna, flag);
    CollisionHeader_GetVirtual(collision, &colHeader);
    this->dyna.bgId = DynaPoly_SetBgActor(play, &play->colCtx.dyna, &this->dyna.actor, colHeader);
```
from experience, doing all of these steps to init dynapoly is pretty annoying, and it being in the macro would be nice. and since it seems like a pretty consistent pattern it seems like it could work. but maybe in practice it wont

Reply:
then the retail version would just omit the error checking

Reply:
I agree on the adding the functions to the macro, and the `s32 pad; CollisionHeader* colHeader = NULL;` is duplicated too so really the whole function is just a macro.

For the string, I don't think it's so bad if we put the mangled version in the header, as long as there's a comment with what the string should be and why

Reply:
The temps are the only thing that can't be added to the macro because of same-line regalloc, i.e.
```c
CollisionHeader* colHeader = NULL; DynaPolyActor_Init(dyn, transformFlags);
```
will break matching. But most of it is fine, updated the branch
https://github.com/zeldaret/oot/compare/main...engineer124:dyna_debug_print?expand=1

Reply:
hmm this is unfortunate. I guess it could have just been copy/pasted between actors but it seems like it should be a macro.
I guess a cursed(?) idea is it was one function that got included into files, but it was marked as static so that it could be named the same thing and not conflict across files. but idk

Reply:
should we revert the changes to this function for these actors and figure it out separately? 
or I guess we can still make the changes now and just macro-ify it later

Reply:
as stated in discord i think we should deal with this later